### PR TITLE
Update ngboost

### PR DESCRIPTION
**Pull request by eduardocorrearaujo**
In this PR, I update some functions related to the ngboost models and add new functions related to the sklearn models and lstm models.

There are examples of the main functions of each new module in the examples folder. I recommend looking into that to help understand the modules that I create.

### CODE REVIEW

**Comment by fccoelho** [comment]
To get started we need some refactoring in the package and module names, for example, the `get_data` module, in this PR refers to pulling data from EpigraphHub, so it should be renamed to `epigraphhub.py`. Since the library can pull data from many sources, we need to keep the names informative. Also, we should avoid hardcoding table  and schema names in the code, such as `colombia`, `casos_posotivos_covid`, we need to write functions  that are completely agnostic to the external environment, and will continue to work if schema and table names change, as long as they get all the required information when they are called.

File: epigraphhub/models/lstm_models.py [hunk 1]
```diff
@@ -262,6 +205,14 @@ def predict(model, Xdata, Ydata, uncertainty=False):
 
 
 def calculate_metrics(pred, ytrue, factor):
+    """
+    Function to return a dataframe with some error metrics computed.
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:209]: Be specific about what metrics are calculated.

File: epigraphhub/models/lstm_models.py [hunk 2]
```diff
@@ -297,29 +248,42 @@ def get_data_model(
...
+    :params ini_date: string. Determines the beggining of the train dataset
+
+    :return: arrays
+    """
+    df = get_cluster_data(
+        "switzerland", predictors, cluster, vaccine=vaccine, smooth=smooth
+    )
 
-    # removing the last three days of data to avoid delay in the reporting.
+    if target_curve_name != "hosp":
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:273]: Still too specific for FOPH dataset.

File: epigraphhub/models/lstm_models.py [hunk 3]
```diff
@@ -422,277 +413,298 @@ def train_eval_canton(
...
     predictors,
-    split=0.75,
-    vaccine=True,
-    smooth=True,
-    ini_date="2020-03-01",
-    updated_data=True,
-    uncertainity=True,
-    save=False,
-    hidden=12,
+    cantons,
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:539]: another specifc parameter name: `cantons`

File: epigraphhub/models/lstm_models.py [hunk 4]
```diff
@@ -422,277 +413,298 @@ def train_eval_canton(
...
 
     """
-    Function to train and evaluate the model for one georegion
+    Function to train and evaluate the model for one georegion.
 
     Important:
-    * By default the function is using the clustering cantons and the data since 2020
     * For the predictor hospCapacity is used as predictor the column ICU_Covid19Patients
+    :params target_curve_name: string to indicate the target column of the predictions
+    :params canton: canton of interest
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:559]: again

File: epigraphhub/models/lstm_models.py [hunk 5]
```diff
@@ -422,277 +413,298 @@ def train_eval_canton(
...
-                indice,
-                batch=1,
-                epochs=epochs,
-                label=f"{target_curve_name}_{canton}",
-                uncertainty=uncertainity,
-                save=save,
-            )
-
-            df_all = pd.concat([df_all, df_pred])
+    all_cantons = [
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:658]: Hardcoding data values. Make this a parameter

File: epigraphhub/models/lstm_models.py [hunk 6]
```diff
@@ -701,216 +713,353 @@ def train_single_canton(
...
-    updated_data=True,
-    uncertainity=True,
-    save=False,
-    path=None,
-    hidden=12,
-    epochs=100,
-    look_back=21,
-    predict_n=14,
-):
+    cantons,
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:716]: again `cantons`

File: epigraphhub/models/lstm_models.py [hunk 7]
```diff
@@ -701,216 +713,353 @@ def train_single_canton(
...
+    :param hidden: number of hidden nodes
+    :params epochs: int. Number of epochs to train the model
+    :params save: boolean. Indicates if the trained model will be saved or not
+    :params path: string. Path where the model will be saved 
+    
+    
+    """
+
+    all_cantons = [
+        "LU",
```
Comment (fccoelho) [inline epigraphhub/models/lstm_models.py:813]: Again hardcoding.

**Review by fccoelho — CHANGES_REQUESTED** [review]
do a complete refactoring of the code to generalize the functions and remove hardcoded swiss data and names.

File: epigraphhub/analysis/clustering.py [hunk 8]
```diff
@@ -82,7 +82,7 @@ def compute_clusters(
     :param t: float. Represent the value used to compute the distance between the clusters
                     and so decide the number of clusters returned.
 
-    :param drop_georegions: list. Param with the georegions that wiil be ignored in the
+    :param drop_georegions: list. Param with the georegions that will be ignored in the
```
[the remark this answers is not in the document]
Comment (eduardocorrearaujo) [reply epigraphhub/analysis/clustering.py:85]: I agree.

**Comment by eduardocorrearaujo** [comment]
The work has migrated to PR 165

**Pull request closed without merging**